src/features/titles/with-test-data.js

```diff
@@ -0,0 +1,118 @@
+/* global wp */
+import deepmerge from 'deepmerge';
+
+const { apiFetch } = wp;
+const { withSelect, withDispatch } = wp.data;
+const { compose, withState } = wp.compose;
+const { __ } = wp.i18n;
+
+const withTestData = compose(
+	withState( {
+		isSaving: false,
+		prevTitles: [],
+		error: false,
+	} ),
+	withSelect( select => {
+		const ab_tests = select( 'core/editor' ).getEditedPostAttribute( 'ab_tests' );
+		const test = ab_tests.titles;
+		const post = select( 'core/editor' ).getCurrentPost();
+		const originalTitles = select( 'core/editor' ).getCurrentPostAttribute( 'ab_test_titles' ) || [];
+		const title = select( 'core/editor' ).getEditedPostAttribute( 'title' ) || '';
+		const titles = select( 'core/editor' ).getEditedPostAttribute( 'ab_test_titles' ) || [];
+		const currentPostType = select( 'core/editor' ).getCurrentPostType();
+		const postType = select( 'core' ).getPostType( currentPostType );
```

Comment:
This feels like it might be nicer just inline in the return unless anything is reused